### PR TITLE
add windows icon

### PR DESCRIPTION
**Pull request by mrbesen**
I added the stuff required to give the exe file an icon.

The problem: It increases the filesize from `69.1kB` to `208.4kB` - Wtf?
I dont know if thats worth for you. Maybe you want to provide both, a icon exe file and an non icon exe file?
It might be possible to reduce the exe filesize by removing some icon resolutions. I don't know which are required.

### CODE REVIEW

**Comment by dreua** [comment]
I don't care about the file size its still probably the smallest application I'd use on windows. But that is on @cnlohr to decide.

The ci fails here, probably needs to run the icon building prior to the exe building. Should be fixable in Makefile.

**Comment by dreua** [comment]
I always think icons are an improvement BTW 👍

**Comment by mrbesen** [comment]
I have removed all icon sizes except 128px and 32px. But i dont know if windows requires some sizes inbetween for correct display. But the binary is now "only" 140kB

**Review by dreua — APPROVED** [review]
Lgtm, not tested in Windows yer

**Comment by cnlohr** [comment]
I don't see the actual .ico file in this CL.

**Comment by dreua** [comment]
> I don't see the actual .ico file in this CL.

Its generated from the svg by Makefile instructions, looks like this. It's good enough to be something distinctive and doesn't look broken :)

**Comment by cnlohr** [comment]
Please also check it in as an artifact

**Comment by dreua** [comment]
I don't see the point in having the ico as a github artifact: @cnlohr you probably just meant checking it in to git as a regular tracked file, didn't you?

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ cnping-wingdi.exe : cnping.c ping.c httping.c resources.o
 cnping.exe : cnping.c ping.c httping.c resolve.c resources.o
 	$(MINGW32)gcc -g -fno-ident -mwindows -m32 -DCNFGOGL $(CFLAGS) -o $@ $^  -lgdi32 -lws2_32 -s -D_WIN32_WINNT=0x0600 -DWIN32 -liphlpapi -lopengl32 -DMINGW_BUILD $(ADMINFLAGS)
 
-resources.o : resources.rc
+resources.o : resources.rc cnping.ico
```
Comment (dreua) [inline Makefile:36]: As we are now committing this file, I'd say we shouldn't require it here to avoid confusion

**Comment by dreua** [comment]
Never mind my comment about artifacts, you already removed that part again in the last commit 👍

File: Makefile (hunk 1 above)
Comment (mrbesen) [reply Makefile]: i have removed it by editing the commit 
from e4c2dc66ed70e9ee3237f9fb6a18569ad8a66134
to [fix ci; reduce amount of resolutions](https://github.com/cntools/cnping/pull/111/commits/07572457a98bb4179fbbb43d85aac5ef47aa319c)

**Comment by cnlohr** [comment]
I am good with this at this point. @dreua do you feel it's ready?

**Comment by dreua** [comment]
Haven't tested it myself but I trust mrbesen's tests, ready to merge from my side.

**Pull request merged**